### ISSUE TITLE
Crash with HttpLoggingInterceptor on old Android devices

### ISSUE THREAD

**Issue by kingsleyadio** [square/okhttp#6079]
OkHttp version: 4.7.0

I found this crash on old Android devices while using OkHttp together with corelibs desugaring on Android Studio 4.0
Wondering if it's a fix that can be done directly in OkHttp, or we may have to wait for this to be supported via desugaring

To reproduce, initialize HttpLoggingInterceptor like:
```kotlin
val interceptor = HttpLoggingInterceptor().setLevel(Level.BODY)
// path to crash is avoided on levels < BODY
```

Possible fix:
Replace `java.nio.charset.StandardCharsets.UTF_8` with `kotlin.text.Charsets.UTF_8`

Relevant stacktrace:
```
     Caused by: java.lang.NoClassDefFoundError: java.nio.charset.StandardCharsets
        at okhttp3.logging.HttpLoggingInterceptor.intercept(HttpLoggingInterceptor.kt:258)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:100)
        at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:34)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:100)
        at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:96)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:100)
        at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:100)
        at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76)
        at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:100)
```

**Comment by yschimke** [comment]
What version of Android was this on?

**Comment by kingsleyadio** [comment]
I tested on Android 4.3

**Comment by yschimke** [comment]
Android 4.3 isn't support with OkHttp 4.7.0 as documented here https://github.com/square/okhttp

Can you try with OkHttp 3.12.12 and reopen if it's still happening.

**Comment by kingsleyadio** [comment]
Hi @yschimke 
Thanks for the quick response

While I do understand this standpoint, I think it will be quite beneficial to _maintain_ some sort of provisional support for these devices especially now that Android Studio 4.0 will include support for most Java 8 APIs as described [here](https://developer.android.com/studio/preview/features/#j8-desugar)

Would you mind taking a look the fix PR #6080 🙏

**Comment by yschimke** [comment]
Absolutely. We have support via the 3.12.x branch which gets security fixes.
